### DISCUSSION TITLE
IgxGrid operator field went missing in filteringExpressionTree

### DISCUSSION THREAD

**Discussion by KaushikEDS** [IgniteUI/igniteui-angular#12484] [Q&A] [answered]
In the igx-grid component, we store the filteringExpressionTree property of the grid in the database to store the filters of each user.
We JSON.stringify the filteringExpressionTree property while storing it in the database.
The searchVal property, which can be a Set sometimes, is converted to Array before storing into the database and vice versa.

This is the function I use to convert the filters : [Link](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABBBA3ApgJygMRgGyiwGUoBDI49KAFTgEFNMyBPANQCYAKYAozUhXQAuRGTAsANMjj4QAWzBUoo5QB4AzlEwwwAcwB8ASkQBvALAAoRDcQxgiHnxLkiAOl6EsuvQHkADljiACYaJhbWtlHAcJiO+NR2iAC8iAAMANxJaoie-ILued76AUFgoW4J+lAAFlkwANQN4VZRbbYoYFq5znGpRQKu6B69PqXM5RoA2jAAuhmt7W32jihyispuNWQaTl6YI+j4wQByZPLoRi2RS0udGNh4+wXoynSMzOzcA9JrCkrUIwLG63RAAX0QRw06DMi1BbQGLxG+zGgQmFQ0-nwMAg6C4MGkAEYgXDQWDSVFySCIVCYSs9vkhm5oWRMBAamwyPhEAAyHk9Z5MllsjlcuxdciQdBwBzKa63RFC9Cs9mc7mpD6sDyYODyBkuITM5UitUk6mkzDUECYJCKoTAsFAA)

So while all this is happening filterState for one user had operator field missing. [Link for filterState](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZgSwDZQKYCcDKUCGaYC8MA3jAESIoYJgDmA8gA4Y5gAmEZAXDANqkVkadDQbN0rDtz4DKw0UxbtOPfuVDsEUBOGkCwOALappZGmQA05BBACqB9AE9pcHEgiorZjQDkjJnjMIAFpzLwALBDY2VDBpKHQAV1QYAF8vRFQkNj9jU0xEgCMASXZE6CdLa1owEHRUAGEcD3ikz3IPHHRgcIA1N2leMgBBNgA3DG0IUSqRjBAIRhxgFIAyGAARVDhYlq9hhHQYADF6hFpw2HWAGRBaG21gTn2qdANtCZgm9BEMZ-JhowlvUkDBhsAVhAIHUEKgIDAbokAB6JJwwADiIBAUn2QKQCGAeB0YBgmBAcCgAHcuiZ9lDULAALKsHC0VDGMBXL7lKDYxwwABCrAA1v8RoleYYQB8UgAlBk4ZCzAX1Sl-ZXoEA4NiE6AzLwCxLINiiGAABU1bESwCgYqahSQa1JeCy+LQsya01q8PW6KM+vIDRAhkMiTABKJ4HhAFEAI5GxgcqAeqMJa3acAwZnyNxijZ4HDmzWQr20BEwegS6GolZsUkYMYEuGzDYPZCO9A+mC9GhQMDqrytioIQoSup5hATTsITJ1gpAuqwTANpt5pJl+W4JVeaNWwkZknL9CNyGzaOOm0iQmgoOGRjgWK28txhNJs8XhLgAlfYP3-ucsVz1QS8v2AGAXwQRNH3LUoKkSJNALARtNTAJM3HLY5lmQLRYXhI8T2bLxjhoVhgAQdDoyRHpWDZLt81wWZjixKlUC6WZ0ViFhQUZDBqM0DxSV5eoxQACVYlBwi+GlNgeERRyE0TxKgSTvhSCCoM5WYxLcZSpPqetj1XLSlJU6T50YfFCPIbSJL0lIABVgPCWokDuZwvBE4MUmKO9NQmJMYE3RUkC06UsnheVoXQJ91gaJIbBSa4aCsshPPKVBwhAbIi2xdNRPg1hArhEAaydaM71cxwAvwoyPMccRDC6YUGS7ecMC+R8B3IUorWHdCgzoVzjAkNAxR6nkRHQv0PDGso+u45ZIn7SovFg1FSJSFUQGaztZlKNBaBG1A60ci9g1DcN92JAyCNm4RlhlGBPOMcDOUmRVUMfPa3okG1JxSHiTULdYatPVafv7LlW3qG0syahlTSC7duohhkbtXGCwDgCQ4JtVFaW6+yf0gRIUFNdZ6GU9rQZS65UBsfGTiwt1cNmJKdlJMjYkhGB7KxdxyxpsVmQMNk6xs3TVNmHjcFBX0kGaeEBq3ZbhZARs4WfH6VbfLxGVJ7RgnxfsYDWiQwBWaWDYQYJlxtOonqyk06DFM1lmFVljpOLEOHLHi8Fdv5wHQi1cptV3wi6RqVglCN3FmUPxCgflYuaRIdP5M2NtmWVgs1bUxXlCjoBddGwfISLWO6STYqja26DLlLl0MAlwD3ITwPjSDdfIZvW7Ke3dq8MkKU2EQwFFWZMGYMi3AQAAvL2BogeDqZXcuyGn4DyJQfkGnCdk47FTBHGgdk8PJKkaVmRyehctyZOHeTxxvpz79ofkRK6NhqX07A6k9uWM01BGAHwkPHLw9kJDOzLLeJY4ZNbrCHAkEcY4h7kCgdaYUAYyAAHVDhZDhPCU6wFzphgjAeRunAAC6aRqFeBAOIPAdRpAAEYMiwmyLkAIvcijjQqM4OhDCmFCTYRwrIOR-D5D4XNBIgj0gkHIHIagdBFDmykKoAQGgTQHj0OQAweRAiHVYsIeykc4heBsPYLozgeCuHcO0bw4BuGmGMS6dAwRlKsDIGkcRXCpGBEaugZqUAmiMFsBANgsxzi1HqJ6HhaZHGdGrv0EKPAACcAAGTJQjyCMJYKIng7ClGcMkYY8gQSQlhIiVE3JggJEuMCAUEosiVrkGTswaQOTUgAFgABQGhoSOgAHSuVoAACmUVgXAaAACUQA)

So my main question being, What can be the reason that operator field went missing for that particular filterState?

**Comment by georgianastasov** [comment, marked as answer]
Hello @KaushikEDS, 

I assume that the missing [operator](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/interfaces/IFilteringExpressionsTree.html#operator) you are talking about is the operator value of the first [filteringOperands](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/interfaces/IFilteringExpressionsTree.html#filteringOperands) array of the [filteringExpressionTree](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#filteringExpressionsTree) property.

I have created a small sample trying to reproduce the described behavior. I am using igx-grid component with filtering feature enabled as [allowFiltering](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#allowFiltering) property set to true and with [qucikFilter](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/filtering#angular-grid-filtering-example) mode. I perform multiple filters on the grid using your custom function for converting the filteringExpressionTree, but still cannot reproduce the described scenarios with the missing operand. However, on my side everything works as expected and I am able to get all operators and all values from filteringExpressionTree.

What I'm assuming is that the described behavior is due to the operation of your custom function and this particular end case of this filterState. Further information on exactly which operator is missing in your case would be helpful.

[Here](https://stackblitz.com/edit/angular-y55qxm?file=src/app/grid/grid-filtering-sample/grid-filtering-sample.component.html) you will find my sample for your reference. Please feel free to modify it with your custom logic of your grid, function and filterState, and send it back to me along with steps to reproduce for further investigation. Alternatively, if the behavir cannot be replicated, please feel free to provide your own sample.
